### DISCUSSION TITLE
X1AC - inverter with waveshare RS485 to Ethernet device - help needed

### DISCUSSION THREAD

**Discussion by Jestur** [wills106/homeassistant-solax-modbus#63] [Q&A] [answered]
Hi all, i have been looking to set this integration for my **solax X1AC** inverter (no PV on this inverter but with 2 x 6.3kWh of solax triple power batteries. However the integration just shows unknown for all the created sensors. inverter serial number begins X1AC362
cloud is connected via pocket wifi. integration version is latest non beta available in HACS 0.4.13a. HA version is core-2022.4.6 running on a pi4, supervisor is 2022.04. have solaredge modbus installed and working OK in case thats relevant.

Process and checks so far
1.) waveshare device on network and configured as per screenshots in gen4 installation guide (guide nice and easy to follow too)
2.) cat5 cable with blue and blue/white pair connected to rs485 A & B - i have checked pin outs on cable and they seem to marry with the correct pin outs (pin 4 and 5 : pin 5 is B pin 4 is A) on waveshare device
3.) integration added through HACS and then added to HA, IP address of waveshare device supplied. baud rate and rs485 address match as per inverter menus.
4.) power led red, link led changes from green to blue on loading the integration. Activity led blinks green only (although hard to tell.
5.) no data coming back to integration - all entities / sensors showing unknown. green activity is approx every 15 seconds as expected, hard to tell if blue receive lights up or not.

things i have tried
1.) swapping pins 4 & 5 in case i am a bit blind - activity light remains lit blue permanently when swapped which seems like a step backwards
2.) double checked all waveshare device settings - although theres not many - protocol is modbus tcp to rtu and advanced settings is simple modbus rtu to tcp. port is 502, DHCP setup and looks ok on router, comms over network to the waveshare device seems fine.
3.) changed baud rating on inverter and waveshare (tried 19200 and 9600) - no difference
4.) tried X1 Gen3 and X1 Gen 4 (i only have self use, force time and backup mode - which seemed to better match some gen3 descriptions)
5.) multiple install and reinstalls of the integration while trying different things - all show no comms back from inverter., a few restarts on HA too for good measure.
6.) physical port on inverter is mark 485 - double checked its the correct one,

any suggestions of what i could have missed ? 

thanks for any help you may be able to provide

J

**Comment by infradom** [comment, marked as answer]
Did you use the most recent beta/prerelease?
The X1AC is currently only supported in the beta version

**Reply by Jestur** [reply]
I hadn't. Just tried 
Main, 0.5.0.b6 and 0.5.0.b8 and the integration fails to setup. (I chose x1 eps option. I don't have the eps (no backup in case of grid failure) functionality on my inverter though. The error reported in the log is : 

This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/solax_modbus/__init__.py:96
Integration: SolaX Inverter Modbus (documentation, issues)
First occurred: 10:24:15 PM (2 occurrences)
Last logged: 10:26:35 PM

Error setting up entry SolaX-modbus for solax_modbus
Error setting up entry SolaX-local for solax_modbus
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 335, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/solax_modbus/__init__.py", line 96, in async_setup_entry
    seriesnumber = decoder.decode_string(14).decode("ascii")
UnboundLocalError: local variable 'decoder' referenced before assignment

Followed by a second error 


This error originated from a custom integration.

Logger: custom_components.solax_modbus
Source: custom_components/solax_modbus/__init__.py:94
Integration: SolaX Inverter Modbus (documentation, issues)
First occurred: 10:24:15 PM (2 occurrences)
Last logged: 10:26:35 PM

cannot perform initial read for serial number


Happy to try and help resolve, but may need some pointers to info you may need.

Hope that helps

J

**Reply by infradom** [reply]
Thanks for this info,
We had to update the code recently to support the AC model, but the error you get seems to be related to failing modbus communicaiton.
It indicates that the initial modbus communication did not succeed, reading the serial number is not possible because the modbus connection could not be opened.

Can you re-verify polarity, baudrate (and ip address of the waveshare)

By the way, if you do not have EPS (emergency power box), you dont have to click any EPS flag (but that should be harmless)

**Reply by Jestur** [reply]
Thx. Not sure what i did differently but working now. I did use a new cat 5 cable so maybe that was it.
All major numbers seem to marry up correctly but happy to go through them all (where possible) if that's useful.
All values currently read only, so hopefully that's something in the pipeline and I'm happy to help testing if that's needed. Particularly interested in mode (self use, force time, back up & feed in priority)

Thx for the help so far and excellent work on the integration.

J

**Reply by infradom** [reply]
Thanks for the update: 
First one warning: afaik we do not have a protocol specification for the writeable parameters of the X1-AC, so we have assumed that the protocol is more or less the same as for the X1 hybrid.
For read-only operations, this is harmless, but using the integration in read-write mode on an X1AC may be dangerous as long as we have no written spec for the writeable parameters.
We however have reports that @mouxy was able to modify the use-mode and another parameter ForceTimeUseP1MaxCapacity.
If you are not able to modify a parameter, it is probably because your inverter is still in 'locked' mode.  
But as I already said, it might be better to wait

**Reply by Mouxy** [reply]
Hi,

My integrations works perfectly for read/write including changing the P1/P2 values with the latest beta.

@infradom Will already has the full documentation (with the write registers) for the AC1X

**Comment by Jestur** [comment]
Hi All, just by way of a quick update. i can confirm the use mode, force time period start, end and max SOC - all appear to do what is required (both reading and writing) see image below. 

Solax web page for remote access on left and test lovelace card on right
![image](https://user-images.githubusercontent.com/77468176/165058674-21ce4d3c-d557-40f8-98ea-27d9bf87a361.png)

Just noticed that the web remote control doesn't include the max SOC (the app does though and that updates correctly)

one strange issue which could be real or not is that the PV 1 and PV 2 power reports around 25 or 26 w (changes values just around that magnitude) - even though the current and voltage for all PV related items are 0 - not a huge issue and could be the inverter reporting these numbers. (NOTE - i dont have any PV connected to the solax so it should be zero) - but thats the only item that looks incorrect/odd as far as i can tell so far.

**Reply by Mouxy** [reply]
The web interface for solax cloud doesn't show max SOC charge but the app after the latest update does.

I believe will is already aware of the PV values showing some readings

But so far I've been using the integration without any problems

**Reply by tomcoleman** [reply]
@wills106 how are you powering the Waveshare RS485 to RJ45 ?

i was thinking about buying https://www.amazon.co.uk/Waveshare-RS485-Ethernet-Converter-auto-negotiation/dp/B09QMNWYLQ/ref=sr_1_3?crid=M776TI484R2S&keywords=Waveshare+RS485+to+ETH&qid=1657544879&sprefix=waveshare+rs485+to+eth%2Caps%2C32&sr=8-3

but didnt know how to power it ?!

**Reply by Mouxy** [reply]
You need a DC power adapter, between 9 and 24V DC

This one will do the job and it has a bit that will let you screw the wires https://amzn.to/3bZ2z45
